pkg/file/builder_test.go

```diff
@@ -2755,3 +2755,93 @@ func Test_getStripPathBasedOnProtocols(t *testing.T) {
 		})
 	}
 }
+
+func Test_stateBuilder_ingestRouteKonnect(t *testing.T) {
+	assert := assert.New(t)
+	rand.Seed(42)
+	type fields struct {
+		currentState *state.KongState
+	}
+	type args struct {
+		route FRoute
+	}
+	tests := []struct {
+		name      string
+		fields    fields
+		args      args
+		wantErr   bool
+		wantState *utils.KongRawState
+	}{
+		{
+			name: "traditional route",
+			fields: fields{
+				currentState: emptyState(),
+			},
+			args: args{
+				route: FRoute{
+					Route: kong.Route{
+						Name: kong.String("foo"),
+					},
+				},
+			},
+			wantErr: false,
+			wantState: &utils.KongRawState{
+				Routes: []*kong.Route{
+					{
+						ID:            kong.String("538c7f96-b164-4f1b-97bb-9f4bb472e89f"),
+						Name:          kong.String("foo"),
+						PreserveHost:  kong.Bool(false),
+						RegexPriority: kong.Int(0),
+						StripPath:     kong.Bool(false),
+						Protocols:     kong.StringSlice("http", "https"),
+					},
+				},
+			},
+		},
+		{
+			name: "expression route",
+			fields: fields{
+				currentState: emptyState(),
+			},
+			args: args{
+				route: FRoute{
+					Route: kong.Route{
+						Name:       kong.String("foo"),
+						Expression: kong.String(`'(http.path == "/test") || (http.path ^= "/test/")'`),
+					},
+				},
+			},
+			wantErr: false,
+			wantState: &utils.KongRawState{
+				Routes: []*kong.Route{
+					{
+						ID:           kong.String("5b1484f2-5209-49d9-b43e-92ba09dd9d52"),
+						Name:         kong.String("foo"),
+						PreserveHost: kong.Bool(false),
+						Expression:   kong.String(`'(http.path == "/test") || (http.path ^= "/test/")'`),
+						Priority:     kong.Uint64(0),
+						StripPath:    kong.Bool(false),
+						Protocols:    kong.StringSlice("http", "https"),
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			b := &stateBuilder{
+				currentState: tt.fields.currentState,
+				isKonnect:    true,
+			}
+			b.rawState = &utils.KongRawState{}
+			d, _ := utils.GetDefaulter(ctx, defaulterTestOpts)
+			b.defaulter = d
+			b.intermediate, _ = state.NewKongState()
```

Comment:
Too late to the party but any particular reason not to check the errors here?

Reply:
could be done, but not meaningful for the specific test...copy&pasta mistake